clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEvent.java

```diff
@@ -19,23 +19,35 @@
 /**
  * This is the abstract definition of the events created by the KafkaConsumer API
  */
-abstract public class ApplicationEvent {
-    public final Type type;
+public abstract class ApplicationEvent {
+
+    public enum Type {
+        NOOP,
+        ASSIGN_PARTITIONS,
+        COMMIT_ASYNC,
```

Comment:
I wonder if we need to treat async and sync commit differently.  I think they look the same from the background thread's prospective.